apps/kg-explorer/src/app/pages/metadata-page/metadata-page.spec.ts

```diff
@@ -0,0 +1,534 @@
+          accessUrl:
+            'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2#2d-ftu-kidney-ascending-thin-loop-of-henle.png',
+          mediaType: 'image/png',
+        },
+        {
+          id: 'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2#2d-ftu-kidney-ascending-thin-loop-of-henle.ai',
+          label: "Raw data distribution '2d-ftu-kidney-ascending-thin-loop-of-henle.ai' file",
+          title: "Raw data distribution '2d-ftu-kidney-ascending-thin-loop-of-henle.ai' file",
+          downloadUrl:
+            'https://cdn.humanatlas.io/digital-objects/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2/assets/2d-ftu-kidney-ascending-thin-loop-of-henle.ai',
+          accessUrl:
+            'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2#2d-ftu-kidney-ascending-thin-loop-of-henle.ai',
+          mediaType: 'application/postscript',
+        },
+        {
+          id: 'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2#crosswalk.csv',
+          label: "Raw data distribution 'crosswalk.csv' file",
+          title: "Raw data distribution 'crosswalk.csv' file",
+          downloadUrl:
+            'https://cdn.humanatlas.io/digital-objects/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2/assets/crosswalk.csv',
+          accessUrl: 'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2#crosswalk.csv',
+          mediaType: 'text/csv',
+        },
+      ],
+    },
+  };
+
+  const mockDoData = {
+    '@graph': [
+      {
+        '@id': 'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle',
+        '@type': 'dcat:Catalog',
+        cell_count: '6',
+        hraVersions: ['v2.3', 'v2.2', 'v2.1', 'v2.0', 'v1.4', 'v1.3'],
+        doVersion: 'v1.2',
+        organs: ['kidney'],
+        organIds: ['http://purl.obolibrary.org/obo/UBERON_0002113'],
+        title: '2D Ascending Thin Limb of Loop of Henle FTU for Kidney',
+        doType: '2d-ftu',
+        lastUpdated: '2024-12-15',
+        doName: 'kidney-ascending-thin-loop-of-henle',
+        versions: ['v1.2', 'v1.1', 'v1.0'],
+        purl: 'https://purl.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle',
+        datasets: [
+          'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2',
+          'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.0',
+          'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.1',
+        ],
+        lod: 'https://lod.humanatlas.io/2d-ftu/kidney-ascending-thin-loop-of-henle/v1.2',
+      },
+      {
+        '@id': 'https://lod.humanatlas.io/ref-organ/heart',
+        versions: ['v1.0', 'v2.0'],
+        organIds: ['heart'],
+        purl: 'https://example.com/purl',
+      },
+    ],
+  };
+
+  it('should navigate to a 404 page if no metadata', async () => {
+    const { fixture } = await render(MetadataPageComponent, {
```

Comment:
Create a small `setup` function that does the common logic instead of repeating the same logic in each test